client/my-sites/media-library/upload-url.jsx

```diff
@@ -13,68 +14,66 @@ import analytics from 'lib/analytics';
 import FormTextInput from 'components/forms/form-text-input';
 import MediaActions from 'lib/media/actions';
 
-export default React.createClass( {
-	displayName: 'MediaLibraryUploadUrl',
+class MediaLibraryUploadUrl extends Component {
+	static propTypes = {
+		className: PropTypes.string,
+		site: PropTypes.object,
+		onAddMedia: PropTypes.func,
+		onClose: PropTypes.func,
+	};
 
-	propTypes: {
-		site: React.PropTypes.object,
-		onAddMedia: React.PropTypes.func,
-		onClose: React.PropTypes.func,
-		className: React.PropTypes.string
-	},
+	static defaultProps = {
+		onAddMedia: noop,
+		onClose: noop,
+	};
 
-	getInitialState() {
-		return {
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
 			value: '',
 			isError: false
 		};
-	},
-
-	getDefaultProps() {
-		return {
-			onAddMedia: noop,
-			onClose: noop
-		};
-	},
+	}
 
-	upload( event ) {
-		var isError = ! event.target.checkValidity();
+	upload = event => {
+		const isError = ! event.target.checkValidity();
 
-		this.setState( {
-			isError: isError
-		} );
+		this.setState( { isError } );
 
 		if ( isError || ! this.props.site ) {
 			return;
 		}
 
+		event.preventDefault();
+
 		MediaActions.clearValidationErrors( this.props.site.ID );
 		MediaActions.add( this.props.site.ID, this.state.value );
 
 		this.replaceState( this.getInitialState() );
```

Comment:
I'm getting a react warning when we're uploading. Let's update this to setState
![screen shot 2017-03-07 at 11 14 01 am](https://cloud.githubusercontent.com/assets/1270189/23673464/3f0328c8-0327-11e7-89c3-6311cf9ae79e.png)